app/views/course/assessment/answer/programming/_test_cases.html.slim

```diff
@@ -10,31 +10,38 @@ h3 = t('.test_cases')
 table.table.table-striped.table-hover
   thead
     tr
-      th = t('.identifier')
-      th = t('.description')
-      th = t('.hint')
+      th = Course::Assessment::Question::ProgrammingTestCase.human_attribute_name(:identifier)
+      th = Course::Assessment::Question::ProgrammingTestCase.human_attribute_name(:description)
+      th = Course::Assessment::Question::ProgrammingTestCase.human_attribute_name(:hint)
       th = t('.passed')
   tbody
     tr
       th colspan=4
         = t('.public')
     - public_test_cases.each do |test_case|
+      - test_case_result = answer_test_results_hash[test_case]
       = content_tag_for(:tr, test_case) do
-        th = test_case.identifier
-        td = test_case.description
-        td = test_case.hint
+        th = format_html(test_case.identifier)
+        td = format_html(test_case.description)
+        td = format_html(test_case.hint) if !test_case_result.try(:passed?)
```

Comment:
Oh, they are not equal ? 

`try` returns `nil` because `test_case_result` is nil, so that the final expression is true, it's same as `!test_case_result`  ?